internal/builder/kiln_metadata.go

```diff
@@ -0,0 +1,54 @@
+package builder
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/crhntr/yamlutil/yamlnode"
+	"gopkg.in/yaml.v3"
+)
+
+type KilnMetadata struct {
+	MetadataGitSHA string `yaml:"metadata_git_sha"`
+	KilnVersion    string `yaml:"kiln_version"`
+}
+
+func setKilnMetadata(in []byte, kilnMetadata KilnMetadata) ([]byte, error) {
+	var productTemplate yaml.Node
+	err := yaml.Unmarshal(in, &productTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse product template: %w", err)
+	}
+
+	_, hasMetadataVersionKey := yamlnode.LookupKey(&productTemplate, "metadata_version")
+	if !hasMetadataVersionKey {
+		return in, nil
+	}
+
+	kilnMetadataValueNode, fieldExists := yamlnode.LookupKey(&productTemplate, "kiln_metadata")
+	if fieldExists {
+		fmt.Println("WARNING: kiln_metadata is not set by kiln please remove it from your product template")
```

Comment:
This error message doesn't make sense to me. What's going on here?

Reply:
If you are adding your own `kiln_metadata` field, it tells you that you don't need to do so because Kiln is adding it now.

Reply:
I added this so

Reply:
How do you feel about the change here: https://github.com/pivotal-cf/kiln/pull/414/commits/f494908d9cf6e50112c47382b1eb6651f7337514